packages/trade-widget/src/components/BuyWidget/BuyWidget.tsx

```diff
@@ -0,0 +1,207 @@
+                        onChange={this._handleTokenSelected.bind(this)}
+                    />
+                </Field>
+                <Label style={{ marginTop: 30 }} isSize="small">
+                    BUY AMOUNT
+                </Label>
+                <Field hasAddons={true}>
+                    <Control isExpanded={true}>
+                        <Input type="text" placeholder="1" onChange={this._handleAmountChange.bind(this)} />
+                    </Control>
+                    <Control>
+                        <Select>
+                            <option>ETH</option>
+                            <option>ZRX</option>
+                        </Select>
+                    </Control>
+                </Field>
+                {/* <Field>
+                    <strong> ESTIMATED COST </strong>
+                </Field> */}
+                <Field style={{ marginTop: 20 }}>
+                    <Button
+                        isLoading={!this._hasQuoteForSelectedToken()}
+                        isFullWidth={true}
+                        isColor="info"
+                        onClick={this._handleSubmitAsync}
+                    >
+                        BUY TOKENS
+                    </Button>
+                </Field>
+                <Field style={{ marginTop: 20 }} isGrouped={'centered'}>
+                    <img style={{ marginLeft: '0px', height: '20px' }} src="/images/powered.png" />
+                </Field>
+            </Content>
+        );
+    }
+
+    private async _handleSubmitAsync(event: any) {
+        event.preventDefault();
+        this.setState((prev, props) => {
+            return { ...prev, isLoading: true };
+        });
+        const { address, quote } = this.props;
+        const { amount } = this.state;
+        const txHash = await this._fillOrderAsync(address, amount, quote.orders[0]);
+        this.setState((prev, props) => {
+            return { ...prev, isLoading: false };
+        });
+    }
+
+    private _handleAmountChange(event: any) {
+        event.preventDefault();
+        const rawValue = event.target.value;
+        let value: undefined | BigNumber;
+        if (!_.isUndefined(rawValue) && !_.isEmpty(rawValue)) {
+            const ethValue = new BigNumber(rawValue);
+            const fillAmount = ZeroEx.toBaseUnitAmount(ethValue, ETH_DECIMAL_PLACES);
+            value = fillAmount;
+            this.setState((prev, props) => {
+                return { ...prev, amount: value };
+            });
+            this._quoteSelectedTokenAsync().catch(console.log);
+        }
+    }
+
+    private _handleTokenSelected(token: AssetToken) {
+        this.props.dispatcher.updateSelectedToken(token);
+    }
+
+    private async _fillOrderAsync(
+        takerAddress: string,
+        fillAmount: BigNumber,
+        signedOrder: SignedOrder,
+    ): Promise<string> {
+        const txHash = await this._getForwarder().fillOrderAsync(signedOrder, fillAmount, takerAddress);
```

Comment:
What happens if the user rejects the tx signing? Network not available? Need error handling.